components/codeChecker/massPreview/Canvas.vue

```diff
@@ -66,12 +66,8 @@ const generateMassPreview = () => {
   }))
 }
 
-useEventListener(window, 'message', async (res) => {
-  const hash = res.data.payload.hash
-  if (
-    res.data?.type === 'kodahash/render/completed'
```

Comment:
why? It's necessary to detect the moment when the rendering process is finished.

Reply:
it's used to calculate the average `time per variation` 

<img src=https://github.com/user-attachments/assets/a4c3d555-184e-4dc2-b340-dd9cce02c08d width=600 />

Reply:
In case some art does not trigger the `complete` event properly, why not calculate the time based on the event triggered?

Reply:
>why not calculate the time based on the event triggered?

I don't understand, the only event that the canvas triggers atm is `kodahash/render/completed` , viki suggested adding a `kodahash/render/started` since some drops took more time to trigger the former but we only have one atm

can you comment some code, maybe that helps me understand better what you mean 

thanks

Reply:
Never mind, I misunderstand it : )